src/util/__tests__/createServerRootMixin.test.js

```diff
@@ -336,187 +374,191 @@ Array [
             },
           },
         },
-        render(h) {
-          return h(InstantSearchSsr, {}, [
-            h(Configure, {
-              attrs: {
-                hitsPerPage: 100,
-              },
-            }),
+        render: renderCompat(h =>
+          h(InstantSearchSsr, {}, [
+            h(
+              Configure,
+              isVue3
+                ? { hitsPerPage: 100 }
+                : {
+                    attrs: {
+                      hitsPerPage: 100,
+                    },
+                  }
+            ),
             h(SearchBox),
-          ]);
-        },
+          ])
+        ),
         serverPrefetch() {
           return this.instantsearch.findResultsState(this);
         },
-      });
+      };
 
-      const wrapper = new Vue({
+      const wrapper = createSSRApp({
         mixins: [forceIsServerMixin],
-        render(h) {
-          return h(App, { props: { someProp } });
-        },
+        render: renderCompat(h =>
+          h(App, isVue3 ? { someProp } : { props: { someProp } })
+        ),
       });
 
       await renderToString(wrapper);
     });
 
-    it('forwards slots', async done => {
-      const searchClient = createFakeClient();
+    // FIXME: make these work with Vue 3
+    if (isVue2) {
```

Comment:
forwarding $slots and $root doesn't work in Vue 3 yet.

However the app (that is created by `createSSRApp()`) doesn't have `$slots` or `$root` as instance variables in the first place. To better understand this issue, I need to find out the real life examples in vue 2 where forwarding `$slots` or `$root` matter, and replicate it with vue 3 to know if it still matters or not.

Reply:
$slots  matter for nuxt where the InstantSearch component is in a layout, $root for similar case with i18n, see https://github.com/algolia/vue-instantsearch/pull/898 for example